x-pack/legacy/plugins/graph/public/components/graph_settings/url_template_form.tsx

```diff
@@ -0,0 +1,316 @@
+            }}
+            onPaste={e => {
+              e.preventDefault();
+              const pastedUrl = e.clipboardData.getData('text/plain');
+              if (isKibanaUrl(pastedUrl)) {
+                setAutoformatUrl(true);
+              }
+              setValue('url', pastedUrl);
+            }}
+            isInvalid={urlPlaceholderMissing || (touched.url && !currentTemplate.url)}
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          fullWidth
+          helpText={currentTemplate.encoder.description}
+          label={i18n.translate('xpack.graph.settings.drillDowns.urlEncoderInputLabel', {
+            defaultMessage: 'URL parameter type',
+          })}
+        >
+          <EuiComboBox
+            fullWidth
+            singleSelection={{ asPlainText: true }}
+            isClearable={false}
+            options={encoders.map(encoder => ({ label: encoder.title, value: encoder }))}
+            selectedOptions={[
+              {
+                label: currentTemplate.encoder.title,
+                value: currentTemplate.encoder,
+              },
+            ]}
+            onChange={choices => {
+              // choices[0].value can't be null because `isClearable` is set to false above
+              setValue('encoder', choices[0].value!);
+            }}
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          fullWidth
+          label={i18n.translate('xpack.graph.settings.drillDowns.toolbarIconPickerLabel', {
+            defaultMessage: 'Toolbar icon',
+          })}
+        >
+          <div role="listbox">
+            {urlTemplateIconChoices.map(icon => (
+              <LegacyIcon
+                key={icon.class}
+                selected={icon === currentTemplate.icon}
+                icon={icon}
+                onClick={() => {
+                  if (currentTemplate.icon === icon) {
+                    setValue('icon', null);
+                  } else {
+                    setValue('icon', icon);
+                  }
+                }}
+              />
+            ))}
+          </div>
+        </EuiFormRow>
+        <EuiFlexGroup justifyContent="flexEnd" responsive={false}>
+          <EuiFlexItem grow={null}>
+            {
+              <EuiButtonEmpty
+                color="danger"
+                onClick={() => {
+                  props.onRemove();
+                }}
+                data-test-subj="graphRemoveUrlTemplate"
+              >
+                {i18n.translate('xpack.graph.settings.drillDowns.removeButtonLabel', {
+                  defaultMessage: 'Remove',
+                })}
+              </EuiButtonEmpty>
+            }
+          </EuiFlexItem>
+          <EuiFlexItem></EuiFlexItem>
+          <EuiFlexItem grow={null}>
+            {
+              <EuiButtonEmpty onClick={reset}>
+                {i18n.translate('xpack.graph.settings.drillDowns.resetButtonLabel', {
+                  defaultMessage: 'Reset',
```

Comment:
This reset button should only appear in a new template form. Otherwise it does nothing if it's in an existing template form.

<img src="https://d.pr/free/i/UYzUYa+" width="50%" />

Reply:
It resets the values to the state before editing - you can use this to undo your current changes because you changed your mind. I thought it would be nice to have as the logic is already there.

We could also get rid of the submit button and auto-update on every change - then it would also be consistent with the other tabs. What do you think?

Reply:
I like the "Update" action being an actual button instead of auto-updating since there's no "undo" option.

If we're going to keep the "Reset" button can you disable it until there are modifications?

Reply:
Done